tests/functional/base_test.go

```diff
@@ -0,0 +1,336 @@
+		err := k8sClient.Get(ctx, name, instance)
+		g.Expect(k8s_errors.IsNotFound(err)).To(BeFalse())
+	}, timeout, interval).Should(Succeed())
+}
+
+func ManilaSchedulerNotExists(name types.NamespacedName) {
+	Consistently(func(g Gomega) {
+		instance := &manilav1.ManilaScheduler{}
+		err := k8sClient.Get(ctx, name, instance)
+		g.Expect(k8s_errors.IsNotFound(err)).To(BeTrue())
+	}, timeout, interval).Should(Succeed())
+}
+
+func ManilaShareNotExists(name types.NamespacedName) {
+	Consistently(func(g Gomega) {
+		instance := &manilav1.ManilaShare{}
+		err := k8sClient.Get(ctx, name, instance)
+		g.Expect(k8s_errors.IsNotFound(err)).To(BeTrue())
+	}, timeout, interval).Should(Succeed())
+}
+
+func SimulateDeploymentReadyWithPods(name types.NamespacedName, networkIPs map[string][]string) {
+	ss := th.GetDeployment(name)
+	for i := 0; i < int(*ss.Spec.Replicas); i++ {
+		pod := &corev1.Pod{
+			ObjectMeta: ss.Spec.Template.ObjectMeta,
+			Spec:       ss.Spec.Template.Spec,
+		}
+		pod.ObjectMeta.Namespace = name.Namespace
+		pod.ObjectMeta.GenerateName = name.Name
+
+		var netStatus []networkv1.NetworkStatus
+		for network, IPs := range networkIPs {
+			netStatus = append(
+				netStatus,
+				networkv1.NetworkStatus{
+					Name: network,
+					IPs:  IPs,
+				},
+			)
+		}
+		netStatusAnnotation, err := json.Marshal(netStatus)
+		Expect(err).NotTo(HaveOccurred())
+		pod.Annotations[networkv1.NetworkStatusAnnot] = string(netStatusAnnotation)
+
+		Expect(k8sClient.Create(ctx, pod)).Should(Succeed())
+	}
+
+	Eventually(func(g Gomega) {
+		ss := th.GetDeployment(name)
+		ss.Status.Replicas = 1
+		ss.Status.ReadyReplicas = 1
+		g.Expect(k8sClient.Status().Update(ctx, ss)).To(Succeed())
+
+	}, timeout, interval).Should(Succeed())
+
+	logger.Info("Simulated deployment success", "on", name)
+}
+
+func SimulateStatefulSetReplicaReadyWithPods(name types.NamespacedName, networkIPs map[string][]string) {
+	ss := th.GetStatefulSet(name)
+	for i := 0; i < int(*ss.Spec.Replicas); i++ {
+		pod := &corev1.Pod{
+			ObjectMeta: ss.Spec.Template.ObjectMeta,
+			Spec:       ss.Spec.Template.Spec,
+		}
+		pod.ObjectMeta.Namespace = name.Namespace
+		pod.ObjectMeta.GenerateName = name.Name
+
+		var netStatus []networkv1.NetworkStatus
+		for network, IPs := range networkIPs {
+			netStatus = append(
+				netStatus,
+				networkv1.NetworkStatus{
+					Name: network,
+					IPs:  IPs,
+				},
+			)
+		}
+		netStatusAnnotation, err := json.Marshal(netStatus)
+		Expect(err).NotTo(HaveOccurred())
+		pod.Annotations[networkv1.NetworkStatusAnnot] = string(netStatusAnnotation)
+
+		Expect(k8sClient.Create(ctx, pod)).Should(Succeed())
+	}
+
+	Eventually(func(g Gomega) {
+		ss := th.GetStatefulSet(name)
+		ss.Status.Replicas = 1
+		ss.Status.ReadyReplicas = 1
+		g.Expect(k8sClient.Status().Update(ctx, ss)).To(Succeed())
+
+	}, timeout, interval).Should(Succeed())
+
+	logger.Info("Simulated statefulset success", "on", name)
+}
+
+func DeleteInstance(instance client.Object) {
```

Comment:
We already have this in lib-common https://github.com/openstack-k8s-operators/lib-common/blob/696d321a942e9617c3049d3f1ae0944f20b9ce58/modules/test/helpers/instance.go#L25